core/txpool/txpool.go

```diff
@@ -356,14 +356,30 @@ func (p *TxPool) Add(txs []*types.Transaction, local bool, sync bool) []error {
 //
 // The transactions can also be pre-filtered by the dynamic fee components to
 // reduce allocations and load on downstream subsystems.
-func (p *TxPool) Pending(filter PendingFilter) map[common.Address][]*LazyTransaction {
-	txs := make(map[common.Address][]*LazyTransaction)
-	for _, subpool := range p.subpools {
-		for addr, set := range subpool.Pending(filter) {
-			txs[addr] = set
+func (p *TxPool) Pending(filter PendingFilter) Pending {
+	var pending Pending
+	// This code is not quite correct, it assumes that the filter requests
+	// _either_ blobs or legacy.
+	if !filter.OnlyBlobTxs && !filter.OnlyPlainTxs {
+		panic("please request either only-blobs or only-plain")
+	}
+	for _, sub := range p.subpools {
+		if pending = sub.Pending(filter); !pending.Empty() {
+			return pending
 		}
 	}
-	return txs
+	return pending
+}
+
+// PendingHashes retrieves the hashes of all currently processable transactions.
+// The returned list is grouped by origin account and sorted by nonce
+func (p *TxPool) PendingHashes(filter PendingFilter) []common.Hash {
```

Comment:
The `Pending` iterates over the subpools. Now, if it gets two non-empty responses of the interface-type `Pending`, it simply cannot do anything, because it is unable to merge the two things. To avoid that, the `panic` is there to prevent a weird bug from being introduced, where we would just discard one or the other and scratch our heads wondering why only one shows up. 

In the `PendingHashes`, however, we _can_ handle it, because it's just two lists of `common.Hash`. We are fully capable of merging those and returning to the caller.